### DISCUSSION TITLE
How to get the previous atom_idx from the two fragments that match with the considered atoms in parent?

### DISCUSSION THREAD

**Discussion by IchiruTake** [rdkit/rdkit#3669] [General] [answered]
Hello!

I am trying to find the atom or atom index in the two fragments that match with the considered atoms in parent after I broke the bond between them. 
At first I create a copied version of the "mol" (TempMol) and separated into two parts which is FragMolA and FragMolB. After breaking the mol, I want to try to get the two atoms of the two fragments above but the atom_index has been changed so that FindAtomEnvironmentOfRadiusN is not worked. Is there a way to access the two atoms in the two fragments or get equivalent result without create two fragments. Thank you 

The below is the source code:
```
    Begin_SubMol, Ending_SubMol = [], []
    if type(Mol) is str:
        Mol = MolFromSmiles(Mol)
    Mol = AddHs(Mol)
    AssignStereochemistry(Mol, flagPossibleStereoCenters=True, force=True)
    Kekulize(Mol, clearAromaticFlags=True)

    bond = Mol.GetBondWithIdx(bond_idx)
    BeginAtom, EndAtom = bond.GetBeginAtom(), bond.GetEndAtom()

    if bond.IsInRing() is False and str(bond.GetBondType()) == 'SINGLE':
        pass
    else:
        raise ValueError("This bond is unable to be break")

    TempMol = RWMol(Mol)
    TempMol.RemoveBond(BeginAtom.GetIdx(), EndAtom.GetIdx())
    TempMol.GetAtomWithIdx(BeginAtom.GetIdx()).SetNoImplicit(True)
    TempMol.GetAtomWithIdx(EndAtom.GetIdx()).SetNoImplicit(True)

    SanitizeMol(TempMol)

    # Convert the two molecules into a SMILES string
    Fragmented_SMILES = str(MolToSmiles(TempMol))
    FragA, FragB = sorted(Fragmented_SMILES.split('.'))
    FragMolA, FragMolB = MolFromSmiles(FragA), MolFromSmiles(FragB)
    
    BeginEnv = FindAtomEnvironmentOfRadiusN(FragMolA, radius=AtomEnvironment, rootedAtAtom=BeginAtom.GetIdx(),
                                            useHs=includeHydrogenEnvironment)
    EndingEnv = FindAtomEnvironmentOfRadiusN(FragMolB, radius=AtomEnvironment, rootedAtAtom=EndAtom.GetIdx(),
                                             useHs=includeHydrogenEnvironment)
    Begin_Mol, End_Mol = PathToSubmol(FragMolA, path=BeginEnv), PathToSubmol(FragMolB, path=EndingEnv)
```

**Comment by greglandrum** [comment, marked as answer]
@IchiruTake: I'm not 100% sure what you want to do, but a relatively easy way to get the environment around an atom after breaking a bond from it is to use `Chem.FragmentOnBonds()` to break the bond(s) you want without changing the atom indices.

Here's a gist demonstrating how that works:
https://gist.github.com/greglandrum/93a209c0155d82f0fde07d14b4a14fbe

Does that help?
